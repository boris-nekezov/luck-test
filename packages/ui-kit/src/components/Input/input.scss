@import '../../styles/app.scss';

.input {
	@include theme-switcher() {
		label {
			color: $text-color--light;
		}

		input {
			color: $text-color--light;
			border: 1px solid $border-color--light;
			background-color: $input-background--light;
		}
	}

	label {
		display: inline-block;
		min-width: 80px;
		margin-right: 24px;
		color: $text-color--dark;
	}

	input {
		padding: 8px 12px;
		color: $text-color--dark;
		border: 1px solid $border-color--dark;
		border-radius: 0;
		outline: 0;
		background-color: $base-background--dark;
	}
}
