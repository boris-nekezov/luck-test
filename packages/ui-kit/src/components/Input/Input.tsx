import './input.scss';

export interface InputProps {
  label: string;
  name: string;
  handleChange: (e: { target: HTMLInputElement }) => void;
  children?: JSX.Element | JSX.Element[],
}

export const Input = (props: InputProps) => {
  const {
    label,
    name,
    handleChange,
    children,
    ...spreadAttributes
  } = props;

  return (
    <div className='input'>
      <label htmlFor={name}>{label}</label>
      <input
        type="text"
        id={name}
        name={name}
        onChange={handleChange}
        {...spreadAttributes}
      />
      {children}
    </div>
  )
}
