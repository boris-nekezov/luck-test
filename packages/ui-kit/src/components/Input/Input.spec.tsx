import React from 'react';
import { render } from '@testing-library/react';
import { Input } from './Input';

describe('Given `Input` component', () => {
  const mockFn = jest.fn();
  let wrapper: ReturnType<typeof render>;


  describe('when it is rendered', () => {
    beforeEach(() => {
      wrapper = render(<Input
        label='User email'
        name='email'
        handleChange={mockFn}
      />);
    });

    it('should match snapshot', () => {
      expect(wrapper.baseElement).toMatchSnapshot();
    })
  })
});
