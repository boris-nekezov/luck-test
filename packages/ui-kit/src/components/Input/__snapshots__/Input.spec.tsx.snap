// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given \`Input\` component when it is rendered should match snapshot 1`] = `
<body>
  <div>
    <div
      class="input"
    >
      <label
        for="email"
      >
        User email
      </label>
      <input
        id="email"
        name="email"
        type="email"
      />
    </div>
  </div>
</body>
`;
